hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordFileParser.java

```
@@ -127,43 +136,33 @@ public static String readPrevFileHash(String fileName) {
         return null;
     }
 
-    private RecordItemParser.INIT_RESULT initFile(String filename) {
-        return recordItemParser.initFile(filename);
-    }
-
-    private void closeFileAndCommit(String fileHash, String previousHash) throws SQLException {
-        recordItemParser.completeFile(fileHash, previousHash);
-    }
-
-    private void rollback() {
-        recordItemParser.rollback();
-    }
-
     /**
      * Given a service record name, read and parse and return as a list of service record pair
      *
-     * @param fileName             the name of record file to read
-     * @param inputStream          input stream of bytes in the record file
-     * @param expectedPrevFileHash the hash of the previous record file in the series
-     * @param thisFileHash         the hash of this file
+     * @param streamFileData containing information about file to be processed
      * @return return boolean indicating method success
      * @throws Exception
      */
-    private boolean loadRecordFile(String fileName, InputStream inputStream, String expectedPrevFileHash,
-                                   String thisFileHash) {
-        var result = initFile(fileName);
-        if (result == RecordItemParser.INIT_RESULT.SKIP) {
-            return true; // skip this fle
-        } else if (result == RecordItemParser.INIT_RESULT.FAIL) {
-            rollback();
+    public boolean loadRecordFile(StreamFileData streamFileData, String thisFileHash, String expectedPrevFileHash) {
+        String fileName = streamFileData.getFilename();
```

any improvements here are welcome.
Will include them in the followup change for '`fileId` removal'.

---

For next PR, I would suggest not returning boolean and let exceptions propagate. The current hash should be loaded within RecordFileParser and passed to it. I would also suggest not keeping previous hash around and passing to loadRecordFile. This can be retrieved from applicationStatus each time. While slightly less efficient, it makes the interface cleaner and the parser more self contained and less coupled.

---

Followup: https://github.com/hashgraph/hedera-mirror-node/pull/587